contracts/RMRK/extension/tokenProperties/RMRKPropertiesAccessControl.sol

```diff
@@ -0,0 +1,251 @@
+    ) external view returns (bool) {
+        return
+            _accessControl[collection][parameterType][parameterId]
+                .specificAddresses[specificAddress];
+    }
+
+    /**
+     * @notice Modifier to check if the caller is authorized to call the function.
+     * @dev If the authorization is set to TokenOwner and the tokenId provided is of the non-existent token, the
+     *  execution will revert with `ERC721InvalidTokenId` rather than `RMRKNotTokenOwner`.
+     * @dev The tokenId parameter is only needed for the TokenOwner authorization type, other authorization types ignore
+     *  it.
+     * @param collection The address of the collection.
+     * @param parameterType The type of the parameter.
+     * @param parameterId The ID of the parameter.
+     * @param tokenId The ID of the token.
+     */
+    modifier onlyAuthorizedCaller(
+        address collection,
+        ParameterType parameterType,
+        uint256 parameterId,
+        uint256 tokenId
+    ) {
+        _onlyAuthorizedCaller(collection, parameterType, parameterId, tokenId);
+        _;
+    }
+
+    /**
+     * @notice Modifier to check if the collection is registered.
+     * @param collection Address of the collection.
+     */
+    modifier onlyRegisteredCollection(address collection) {
+        if (!_issuerSettings[collection].registered) {
+            revert RMRKCollectionNotRegistered();
+        }
+        _;
+    }
+
+    /**
+     * @notice Modifier to check if the collection is not registered.
+     * @param collection Address of the collection.
+     */
+    modifier onlyUnregisteredCollection(address collection) {
+        if (_issuerSettings[collection].registered) {
+            revert RMRKCollectionAlreadyRegistered();
+        }
+        _;
+    }
+
+    /**
+     * @notice Modifier to check if the caller is the issuer of the collection.
+     * @param collection Address of the collection.
+     */
+    modifier onlyIssuer(address collection) {
+        if (_issuerSettings[collection].useOwnable) {
+            if (Ownable(collection).owner() != msg.sender) {
+                revert RMRKNotCollectionIssuer();
+            }
+        } else if (_issuerSettings[collection].issuer != msg.sender) {
+            revert RMRKNotCollectionIssuer();
+        }
+        _;
+    }
+
+    /**
+     * @notice Function to check if the caller is authorized to mamage a given parameter.
+     * @dev The valid parameter types are:
+     *  [
+     *      STRING,
+     *      ADDRESS,
+     *      BYTES,
+     *      UINT,
+     *      BOOL
+     *  ]
+     * @param collection The address of the collection.
+     * @param parameterType The type of the parameter.
+     * @param parameterId The ID of the parameter.
+     * @param tokenId The ID of the token.
+     */
+    function _onlyAuthorizedCaller(
+        address collection,
+        ParameterType parameterType,
+        uint256 parameterId,
+        uint256 tokenId
+    ) private view {
```

Comment:
This looks a lot like what I expected. Just without the parameter type.
I see here why you're using Id on the access control, since you make them as independent as possible. TBH I would just merge them so I could use getIdForKey inside the access control configuration and not have the user figure out what the id even means in that context.